### PR TITLE
fix: set all optional list attrs to have a default value of [] instead of None

### PR DESCRIPTION
**Pull request by mAxYoLo01**
## About

To avoid unnecessary `if not self.list: self.list = []`, the default value of all optional attributes will have a default value of an empty list instead.

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [ ] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [ ] New feature/enhancement
  - [x] Bugfix

### CODE REVIEW

File: interactions/api/models/guild.py [hunk 1]
```diff
@@ -2269,12 +2269,12 @@ class GuildPreview(DictSerializerMixin):
...
 
     id: Snowflake = field(converter=Snowflake)
-    emojis: Optional[List[Emoji]] = field(converter=convert_list(Emoji), default=None)
+    emojis: Optional[List[Emoji]] = field(converter=convert_list(Emoji), factory=list)
     name: str = field()
     icon: Optional[str] = field(default=None)
     splash: Optional[str] = field(default=None)
     discovery_splash: Optional[str] = field(default=None)
-    features: Optional[List[str]] = field(default=None)
+    features: Optional[List[str]] = field(default=[])
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]: Factory

**Review by sillyElvy — CHANGES_REQUESTED** [review]

**Comment by AstreaTSS** [comment]
This is both absolutely not how `Optional` works *and* a breaking change. While I'm okay with it personally, this is something that needs a bit more discussion than it seems.

**Comment by sillyElvy** [comment]
> This is both absolutely not how `Optional` works *and* a breaking change. While I'm okay with it personally, this is something that needs a bit more discussion than it seems.

Does optional enforce `None`? I thought an emtpy list would be fine since `not []` = `True`

**Comment by AstreaTSS** [comment]
> > This is both absolutely not how `Optional` works *and* a breaking change. While I'm okay with it personally, this is something that needs a bit more discussion than it seems.
> 
> Does optional enforce `None`? I thought an emtpy list would be fine since `not []` = `True`

`Optional` is explicitly `Union[X, None]`, yeah. These are no longer `Optional`s - they're just lists.

**Comment by sillyElvy** [comment]
> > > This is both absolutely not how `Optional` works *and* a breaking change. While I'm okay with it personally, this is something that needs a bit more discussion than it seems.
> > 
> > Does optional enforce `None`? I thought an emtpy list would be fine since `not []` = `True`
> 
> `Optional` is explicitly `Union[X, None]`, yeah. These are no longer `Optional`s - they're just lists.

I see, alright

But imho it is certainly better than `if x is None: x = []` in every helper that adds/removes from the list

**Pull request closed without merging**